lab-anthony/app/components/app-gallery/app-gallery-controller.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const angular = require('angular');
+
+angular.module('galleryApp')
+.controller('AppGalleryController', [AppGalleryController]);
+
+function AppGalleryController() {
+
+  this.views = {
+    thumbnail: false,
+    list: true,
+    tile: false
+  };
+
+  this.thumbnailView = function() {
+    this.views.thumbnail = true;
+    this.views.list = false;
+    this.views.tile = false;
+  };
+
+  this.listView = function() {
+    this.views.thumbnail = false;
+    this.views.list = true;
+    this.views.tile = false;
+  };
+
+  this.tileView = function() {
+    this.views.thumbnail = false;
+    this.views.list = false;
+    this.views.tile = true;
+  };
+
+  this.displayModalImage = false;
+
+  this.showModalImage = function(imageUrl) {
+    this.displayImageUrl = imageUrl;
+    this.displayModalImage = true;
+  };
+
+  this.hideModalImage = function() {
+    this.displayModalImage = false;
+  };
+
+  this.images = [
```

Comment:
It would probably be more readable if you broke this out into a separate file and required it in.